src/fs.rs

```diff
@@ -151,6 +154,130 @@ pub fn create_workspace_archive(working_root_dir: &str, execution_id: &str) -> R
     }
 }
 
+pub fn create_yaml_backup_file<P>(
+    working_root_dir: P,
+    chart_name: String,
+    resource_name: String,
+    content: String,
+) -> Result<String, std::io::Error>
+where
+    P: AsRef<Path>,
+{
+    info!("creating yaml backup file for {}", resource_name);
+
+    let dir = working_root_dir.as_ref().join("backups");
+
+    create_dir_all(&dir)?;
+
+    let root_path = dir
+        .to_str()
+        .map(|e| e.to_string())
+        .ok_or_else(|| Error::from(ErrorKind::NotFound));
+
+    let string_path = format!("{}/{}_{}_backup.yaml", root_path?, chart_name, resource_name);
+    let str_path = string_path.as_str();
+    let path = Path::new(str_path);
+
+    let mut file = match File::create(&path) {
+        Err(e) => return Err(e),
+        Ok(file) => file,
+    };
+
+    match file.write(content.as_bytes()) {
+        Err(e) => Err(e),
+        Ok(_) => {
+            file.flush()?;
+            Ok(path
+                .to_str()
+                .map(|e| e.to_string())
+                .ok_or_else(|| Error::from(ErrorKind::NotFound))?)
+        }
+    }
+}
+
+pub fn remove_lines_starting_with(path: String, starter: &str) -> Result<String, std::io::Error> {
+    info!("editing yaml backup file {}", path);
+
+    let mut file = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .open(path.as_str())
+        .expect("file doesn't exist");
+
+    let content = BufReader::new(file.try_clone().unwrap())
+        .lines()
+        .filter(|line| !line.as_ref().unwrap().contains(starter))
+        .map(|line| line.unwrap())
+        .collect::<Vec<String>>()
+        .join("\n");
+
+    file.flush()?;
```

Comment:
no need to flush, you are only reading the file. there is nothing to commit/flush to disk